agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -671,43 +679,27 @@ public static class ConfigurationException extends RuntimeException {
     }
   }
 
-  static Configuration getConfigurationFromConfigFile(Path configPath, boolean strict)
-      throws IOException {
-    ObjectMapper mapper = new ObjectMapper();
-    try (InputStream in = Files.newInputStream(configPath)) {
-      if (!strict) {
-        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-      }
-      return mapper.readValue(in, Configuration.class);
-    } catch (UnrecognizedPropertyException ex) {
-      if (strict) {
-        Configuration configuration = getConfigurationFromConfigFile(configPath, false);
-        configurationLogger.warn(
-            getJsonEncodingExceptionMessageForFile(configPath, ex.getMessage()));
-        return configuration;
-      } else {
-        throw new FriendlyException(
-            getJsonEncodingExceptionMessageForFile(configPath, ex.getMessage()),
-            "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
-      }
-    } catch (JsonMappingException | JsonParseException ex) {
-      throw new FriendlyException(
-          "Error parsing configuration from file: "
-              + configPath.toAbsolutePath()
-              + System.lineSeparator()
-              + System.lineSeparator()
-              + ex.getMessage(),
-          "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358",
-          ex);
-    } catch (Exception e) {
-      throw new ConfigurationException(
-          "Error parsing configuration from file: " + configPath.toAbsolutePath(), e);
+  private static class ConfigurationOrigin {
```

Comment:
`JsonConfiguration` or `JsonConfigDescription` would be more explicit?

Reply:
Renamed to `JsonOrigin`